### DISCUSSION TITLE
FFMPEG bitrate parameter seems to be ignored in some cases

### DISCUSSION THREAD

**Discussion by lakimens** [photoprism/photoprism#2499] [General]
Reference: https://www.reddit.com/r/photoprism/comments/vqgsxu/photoprism_does_not_seem_to_respect_photoprism/

Some converted videos come out close to the original bitrate, with the difference being only H265 → H264 

In the example provided on Reddit:

Original file has a bitrate of 53.5 Mbit/s, and the converted one has a bitrate of 53.4 Mbit/s.

It mostly happens with 4K videos, might be related to high bitrate / 60FPS.

[Files available here.](https://drive.visiting.mk/index.php/s/XWp5byryxFAYRcn?)

**Comment by graciousgrey** [comment]
Thank you for reporting and creating the discussion :)

**Reply by lastzero** [reply]
Quick feedback from my side:

1. The ultimate bitrate may depend on the particular encoder, especially since the exact quality parameters vary and depend on what the encoder actually supports.
2. Since some encoders use additional quality presets and quality-based VBR, I'm not surprised that 4K videos cause FFmpeg to use a higher bitrate, considering that the quality otherwise drops significantly with the 10MBit/s limit you (seem to have) configured.

@lakimens Could you share more details or copy them from Reddit so we can get a complete picture?

You are welcome to look into it in more detail, for example by reading the technical documentation or the source code of the encoder:

- https://code.videolan.org/videolan/x264/-/blob/master/x264.c#L733

FFmpeg encoder parameters:

https://github.com/photoprism/photoprism/blob/bbc4f2f276d54b3f5397dbae3f8b79b4ebcc5b1e/internal/ffmpeg/convert.go#L137


Also check out the logs from FFmpeg when you run it manually. This is great for learning more and experimenting.

**Comment by lakimens** [comment]
Oh, I thought I included enough from Reddit, but I guess I missed something.

- In the command, I can see that the b:v parameter is set correctly to what I've configured (10M)
- 1080P videos and videos under 10M (my set limit) behave much better

There isn't much more in the Reddit thread. I might test more during the weekend if I have the time.
